server/app/jobs/stack_deploy_worker.rb

```diff
@@ -0,0 +1,76 @@
+class StackDeployWorker
+  include Celluloid
+  include Logging
+  include Stacks::SortHelper
+
+  def perform(stack_deploy_id, stack_rev_id)
+    stack_deploy = StackDeploy.find_by(id: stack_deploy_id)
+    stack_rev = StackRevision.find_by(id: stack_rev_id)
+    if stack_deploy && stack_rev
+      deploy_stack(stack_deploy, stack_rev)
+    end
+  end
+
+  # @param [StackDeploy] stack_deploy
+  # @param [StackRevision] stack_rev
+  def deploy_stack(stack_deploy, stack_rev)
+    stack = stack_deploy.stack
+    stack_deploy.ongoing!
+
+    remove_services(stack, stack_rev)
+    services = sort_services(stack.grid_services.to_a)
+    services.each do |service|
+      deploy_service(service, stack_deploy)
+    end
+    stack_deploy.success!
+
+    stack_deploy
+  rescue
+    stack_deploy.error!
```

Comment:
Could we somehow get the possibile service deployment error messages appended here?